pkg/plugins/runtime/k8s/controllers/gatewayapi/http_route_controller.go

```diff
@@ -83,32 +84,65 @@ func (r *HTTPRouteReconciler) gapiToKumaRoutes(
 	ParentConditions,
 	error,
 ) {
+	routeNs := kube_core.Namespace{}
+	if err := r.Client.Get(ctx, kube_types.NamespacedName{Name: route.Namespace}, &routeNs); err != nil {
+		if kube_apierrs.IsNotFound(err) {
+			return nil, nil, nil
+		} else {
+			return nil, nil, err
+		}
+	}
+
 	var refRoutes []gatewayapi.ParentRef
+	conditions := ParentConditions{}
 
 	kumaRoute := &mesh_proto.GatewayRoute{}
 
 	// Convert GAPI parent refs into selectors
 	for i, ref := range route.Spec.ParentRefs {
-		if handle, err := r.shouldHandleParentRef(ctx, route, ref); err != nil {
+		refAttachment, err := attachment.ParentRefAttachment(ctx, r.Client, &routeNs, ref)
+		if err != nil {
 			return nil, nil, errors.Wrapf(err, "unable to check parent ref %d", i)
-		} else if !handle {
-			continue
 		}
-		refRoutes = append(refRoutes, ref)
 
-		refNamespace := route.Namespace
-		if ns := ref.Namespace; ns != nil {
-			refNamespace = string(*ns)
+		switch refAttachment {
+		// TODO are these three cases reflected correctly?
+		// specifically does ResolvedRefs cover ParentRefs or only BackendRefs?
+		case attachment.NotPermitted, attachment.Invalid:
+			var message string
+			switch refAttachment {
+			case attachment.NotPermitted:
+				message = "attachment to parent not permitted by AllowedRoutes"
+			case attachment.Invalid:
+				// TODO missing a specific Reason for this?
+				message = "reference to parent is invalid"
+			}
+			conditions[ref] = []kube_meta.Condition{
+				{
+					Type:    string(gatewayapi.ConditionRouteResolvedRefs),
```

Comment:
This is what my first thought was but wasn't sure cause the ParentRef is... also a ref. So you're thinking we shouldn't set ResolvedRefs here but instead only set `ConditionRouteAccepted` to `Refused` and explain in the message.

Reply:
It seems like there's a soft convention to have the "overall" status (`Ready` or here `Accepted`) only have a "yes" and "no" reason and have additional conditions to reflect why the main condition is false. Which sort of pushed me into thinking `ResolvedRefs` would fit here (because there's only `Accepted` and `ResolvedRefs` at the moment for routes)